packages/cactus-test-tooling/src/test/typescript/integration/common/containers.test.ts

```diff
@@ -12,114 +12,118 @@ import {
-  t.end();
+  test(testCase, async () => {
+    const srcFileName = uuidV4();
+    const srcFileDir = os.tmpdir();
+    const dstFileDir = "/";
+    const dstFileName = srcFileName;
+    const srcFilePath = path.join(srcFileDir, srcFileName);
+    const dstFilePath = path.join(dstFileDir, dstFileName);
+
+    const fileContents = {
+      id: srcFileName,
+      message: "Hello world!",
+    };
+
+    const srcFileAsString = JSON.stringify(fileContents);
+    fs.writeFileSync(srcFilePath, srcFileAsString);
+    {
+      const res: IncomingMessage = await Containers.putFile({
+        containerOrId: container,
+        srcFileDir,
+        srcFileName,
+        dstFileDir,
+        dstFileName,
+      });
+
+      expect(res).toBeTruthy();
+      expect(typeof res.statusCode).toBe("number");
+      const statusCode: number = res.statusCode as number;
+
+      expect(statusCode).toBeGreaterThan(199);
+      expect(statusCode).toBeLessThan(300);
+      expect(res.statusMessage).toEqual("OK");
+
+      log.debug("Put file result: %o %o", res.statusCode, res.statusMessage);
+
+      const fileAsString2 = await Containers.pullFile(container, dstFilePath);
+      expect(fileAsString2).toBeTruthy();
+
+      const fileContents2 = JSON.parse(fileAsString2);
+      expect(fileContents2).toBeTruthy();
+      expect(fileContents2.id).toEqual(fileContents.id);
+    }
+
+    {
+      expect(httpEchoContainer).toBeTruthy();
+      const container = await httpEchoContainer.start();
+      expect(container).toBeTruthy();
+
+      const diag = await Containers.getDiagnostics({ logLevel: "TRACE" });
+      expect(diag).toBeTruthy();
+
+      expect(diag.containers).toBeTruthy();
+      expect(Array.isArray(diag.containers)).toBeTruthy();
+      expect(diag.containers.length > 0).toBeTruthy();
+
+      expect(diag.images).toBeTruthy();
+      expect(diag.images.length > 0).toBeTruthy();
+      expect(Array.isArray(diag.images)).toBeTruthy();
+
+      expect(diag.info).toBeTruthy();
+
+      expect(diag.networks).toBeTruthy();
+      expect(diag.networks.length > 0).toBeTruthy();
+      expect(Array.isArray(diag.networks)).toBeTruthy();
+
+      expect(diag.version).toBeTruthy();
+
+      expect(diag.volumes).toBeTruthy();
+      expect(diag.volumes.Volumes).toBeTruthy();
+      expect(Array.isArray(diag.volumes.Volumes)).toBeTruthy();
+    }
+
+    {
+      const badSocketPath = "/some-non-existent-path/to-make-it-trip-up/";
+      try {
+        await Containers.getDiagnostics({
+          logLevel: "TRACE",
+          // pass in an incorrect value for the port so that it fails for sure
+          dockerodeOptions: {
+            port: 9999,
+            socketPath: badSocketPath,
+          },
+        });
+        fail(Containers.getDiagnostics);
+      } catch (ex) {
+        expect(ex).toBeTruthy();
+        expect(ex.cause).toBeTruthy();
+        expect(ex.cause.message).toBeTruthy();
+        const causeMsgIsInformative = (ex as any).cause.message.includes(
```

Comment:
@Leeyoungone Could you refactor this to not use `any`? I'm thinking of something like what they are showing off here: https://github.com/facebook/jest/issues/7284#issuecomment-433426694

Reply:
Hello Peter! Thanks for the link, I went ahead and implemented this test to use `toHaveProperty`for this.

`connect ENOENT /some-non-existent-path/to-make-it-trip-up/`

However, to use the `toHaveProperty` I had to add two more lines of code (shown below) 

``` typescript
const connectEN = "connect ENOENT ";
const fullBadPath = connectEN.concat(badSocketPath);
expect(ex).toHaveProperty(["cause", "message"], fullBadPath);
```
and this is because the path `ex.cause.message` had `connect ENOENT /some-non-existent-path/to-make-it-trip-up/` instead of just `/some-non-existent-path/to-make-it-trip-up/`. Using the `toHaveProperty` does work, it's just a few more lines. 

Another way to do this with less lines by using `toMatch`, here it is below 

``` typescript 
expect(ex.cause.message).toMatch(badSocketPath);
```
Both of them work, I just wanted to ask which one you'd prefer before I refactor this test case and push again!!

Reply:
@Leeyoungone Sorry for the very slow response! The simpler one liner[1] should work fine in this case because as long as the message does contain the socket path, we are okay (so we don't need an exact match).

[1] `expect(ex.cause.message).toMatch(badSocketPath);`

Reply:
> @Leeyoungone Sorry for the very slow response! The simpler one liner[1] should work fine in this case because as long as the message does contain the socket path, we are okay (so we don't need an exact match).
> 
> [1] `expect(ex.cause.message).toMatch(badSocketPath);`

@Leeyoungone Sorry, nvm, I just checked it and it won't compile with the second option when the exception type is defined as `unknown` so better to go with the first option. I made the changes quickly just now so you don't have to worry about it, no action necessary. Thank you again!